lib/iris/cube.py

```diff
@@ -2892,6 +2892,10 @@ def _xml_element(self, doc, checksum=False, order=True, byteorder=True):
         if self.var_name:
             cube_xml_element.setAttribute('var_name', self.var_name)
         cube_xml_element.setAttribute('units', str(self.units))
+        if self.fill_value is not None:
+            cube_xml_element.setAttribute('fill_value', str(self.fill_value))
+        cube_xml_element.setAttribute('dtype', self.dtype.name)
+        cube_xml_element.setAttribute('core-dtype', self.core_data.dtype.name)
```

Comment:
Add the `cube` level `fill_value` and `dtype` and interim `core-dtype`, which will be refactored away later but greatly assists with clarity at this moment